gsy_e_sdk/setups/Asset_API_scripts/rest_bid_offer_posting.py

```diff
@@ -0,0 +1,158 @@
+        """Place a bid or an offer whenever a new market is created."""
+        if self.is_finished is True:
+            return
+        self.post_bid_offer()
+
+    def on_tick(self, tick_info):
+        pass
+
+    def post_bid_offer(self):
+        """Post a bid or an offer to the exchange."""
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            if "asset_info" not in area_dict or area_dict["asset_info"] is None:
+                continue
+
+            # Consumption assets
+            if key_in_dict_and_not_none_and_greater_than_zero(
+                area_dict["asset_info"], "energy_requirement_kWh"
+            ):
+                energy = area_dict["asset_info"]["energy_requirement_kWh"]
+                self.add_to_batch_commands.bid_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=energy
+                )
+
+            # Generation assets
+            if key_in_dict_and_not_none_and_greater_than_zero(
+                area_dict["asset_info"], "available_energy_kWh"
+            ):
+                energy = area_dict["asset_info"]["available_energy_kWh"]
+                self.add_to_batch_commands.offer_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=energy
+                )
+
+            # Storage assets
+            if key_in_dict_and_not_none_and_greater_than_zero(
+                area_dict["asset_info"], "energy_to_buy"
+            ):
+                buy_energy = area_dict["asset_info"]["energy_to_buy"]
+                self.add_to_batch_commands.bid_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=buy_energy
+                )
+
+                sell_energy = area_dict["asset_info"]["energy_to_sell"]
+                self.add_to_batch_commands.offer_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=sell_energy
+                )
+
+            self.execute_batch_commands()
+
+    def on_event_or_response(self, message):
+        pass
+
+    def on_finish(self, finish_info):
+        self.is_finished = True
+
+
+def get_assets_name(indict: dict) -> dict:
+    """
+    Parse the grid tree and return all registered assets
+    wrapper for _get_assets_name
+    """
+    if indict == {}:
+        return {}
+    outdict = {"Area": [], "Load": [], "PV": [], "Storage": []}
+    _get_assets_name(indict, outdict)
+    return outdict
+
+
+def _get_assets_name(indict: dict, outdict: dict):
+    """
+    Parse the Collaboration / Canary Network registry
+    Return a list of the Market nodes the user is registered to
+    """
+    for key, value in indict.items():
+        if key == "name":
+            name = value
+        if key == "type":
+            area_type = value
+        if key == "registered" and value:
+            outdict[area_type].append(name)
+        if "children" in key:
+            for children in indict[key]:
+                _get_assets_name(children, outdict)
```

Comment:
As far as I understand, this method takes the name of the area and, if it's registered, adds it to `outdict` (which is a dict of all the registered assets).

I know this implementation already existed, but I think it would be good to refactor it because the way this works now is a bit obscure. Would this be the same?
```suggestion
    def _get_assets_name(indict: dict, outdict: dict):
        if indict.get("registered") is True:
            area_type = indict["type"]
            outdict[area_type].append(indict["name"])

        for child in indict.get("children", []):
            _get_assets_name(child, outdict)
```
I gave for granted that we always have `area_type` inside.

Do you think this is equivalent to the other version? Tagging @hannesdiedrich to also ask for his opinion.

(I would also suggest to rename `indict` to `node` or `area` and `outdict` to `registered_assets`)